provisioning/plugins/common/unsignedcorim_decoder.go

```diff
@@ -43,7 +47,20 @@ func UnsignedCorimDecoder(data []byte, xtr IExtractor) (*decoder.EndorsementDeco
 		return nil, fmt.Errorf("invalid unsigned corim: %w", err)
 	}
 
-	// TODO(tho) check profile
+	if uc.Profiles != nil {
+		// get the profile
+		if len(*uc.Profiles) > 1 {
+			return nil, fmt.Errorf("invalid multiple profiles: %x", len(*uc.Profiles))
+		}
+		p := (*uc.Profiles)[0]
+		profile, err := p.Get()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get the profile information: %w", err)
+		}
+		xtr.SetProfile(profile)
+	} else {
+		xtr.SetProfile(noProfile)
```

Comment:
is it acceptable not having a profile?

Reply:
Profile in CoRIM is an Optional Field.

We resolve everything via MediaType as of today! From MediaType we move to PSA Decoder.

For now assumption is that if CoRIM profile is missing, then we resort to PSA Profile! 
We have set CoRIM Profile to be populated mandatory for CCA

Reply:
> For now assumption is that if CoRIM profile is missing, then we resort to PSA Profile!

this doesn't look right